chain/src/chain.rs

```diff
@@ -271,24 +271,23 @@ impl Chain {
 		res
 	}
 
-	fn determine_status(&self, head: Option<Tip>, prev_head: Tip) -> BlockStatus {
-		// We have more work if the chain head is updated.
-		let is_more_work = head.is_some();
-
-		let mut is_next_block = false;
-		let mut reorg_depth = None;
+	fn determine_status(&self, head: Option<Tip>, prev_head: Tip, fork_point: Tip) -> BlockStatus {
+		// If head is updated then we are either "next" block or we just experienced a "reorg" to new head.
+		// Otherwise this is a "fork" off the main chain.
 		if let Some(head) = head {
 			if head.prev_block_h == prev_head.last_block_h {
-				is_next_block = true;
+				BlockStatus::Next { prev_head }
 			} else {
-				reorg_depth = Some(prev_head.height.saturating_sub(head.height) + 1);
```

Comment:
Pretty sure this is some questionable logic.
`prev_head` is _always_ a lower height than `head`, even in a fork/reorg scenario, except for some unlikely edge cases.
So we basically always saturating_sub to zero and then add 1, so we always see `depth = 1`.

This is incorrect behavior - we should not be comparing `prev_head` to `head` (or `head` to `prev_head`). What we want to be doing to find the "depth" is to look back to the "fork point" that we rewound to to switch from the current chain to the new chain. The "fork point" is the common ancestor of both `head` and `prev_head` and is where the chains diverge. A "deep" reorg will diverge earlier than a "shallow" reorg.